clients/src/main/resources/common/message/OffsetCommitRequest.json

```diff
@@ -31,13 +31,19 @@
   // version 7 adds a new field called groupInstanceId to indicate member identity across restarts.
   //
   // Version 8 is the first flexible version.
-  "validVersions": "0-8",
+  //
+  // Version 9 is the first version that can be used with the new consumer group protocol (KIP-848). The
+  // request is the same as version 8.
+  // Version 9 is added as part of KIP-848 and is still under development. Hence, the last version of the
+  // API is not exposed by default by brokers unless explicitly enabled.
+  "latestVersionUnstable": true,
+  "validVersions": "0-9",
   "flexibleVersions": "8+",
   "fields": [
     { "name": "GroupId", "type": "string", "versions": "0+", "entityType": "groupId",
       "about": "The unique group identifier." },
-    { "name": "GenerationId", "type": "int32", "versions": "1+", "default": "-1", "ignorable": true,
-      "about": "The generation of the group." },
+    { "name": "GenerationIdOrMemberEpoch", "type": "int32", "versions": "1+", "default": "-1", "ignorable": true,
+      "about": "The generation of the group if the generic group protocol or the member epoch if the consumer protocol." },
```

Comment:
The new group coordinator uses the member epoch and the old one uses the generation id I believe.

Reply:
It based on the type of the group. In the new group coordinator, we have two types: generic (the old protocol) and consumer (the new protocol).

Reply:
> is it safe to change the field name?

Yes, it is as this does not change the serialized bytes.

> you're saying based on the consumer since we can have a group with both new & old consumers right?

It is really based on the group type. If we have a group with both old and new consumers, it means that the group is being upgraded so the new "engine" is used for both in this case and the member epoch is the expected value for both as well. Of course, the old consumers won't really know about this. They will believe that they send the generation id.

> nit: "if using ... if using ..."

Ack.